### PR TITLE
[documentation] Give more details on coloring algorithms

### PR DESCRIPTION
**Pull request by amontoison**
Related to #246
I propose to use the most efficient coloring for `SparseReverseADHessian`.

### CODE REVIEW

File: docs/src/sparse.md [hunk 1]
```diff
@@ -38,7 +38,16 @@ The default detector is `TracerSparsityDetector()` from the package `SparseConne
...
 - A **`coloring_algorithm`** must be of type `SparseMatrixColorings.GreedyColoringAlgorithm`.
-The default algorithm is `GreedyColoringAlgorithm{:direct}()` from the package `SparseMatrixColorings.jl`.
+The default algorithm is `GreedyColoringAlgorithm{:direct}()` for `SparseADJacobian` and `SparseADHessian`, while it is `GreedyColoringAlgorithm{:substitution}()` for `SparseReverseADHessian`.
+These algorithms are available in the package `SparseMatrixColorings.jl`.
+
+The `GreedyColoringAlgorithm{:direct}()` performs column coloring for Jacobians and star coloring for Hessians.
+In contrast, `GreedyColoringAlgorithm{:substitution}()` applies acyclic coloring for Hessians.
+Substitution coloring results in fewer colors and thus requires fewer directional derivatives to recover all non-zeros of the sparse Hessian.
+However, it necessitates storing the entire compressed sparse Hessian.
+On the other hand, direct coloring only requires storing one column of the compressed Hessian at a time.
```
Comment (tmigot) [inline docs/src/sparse.md:48]:
```suggestion
However, it requires storing the compressed sparse Hessian, while `:direct` coloring only stores one column of the compressed Hessian.
```

File: docs/src/sparse.md [hunk 2]
```diff
@@ -38,7 +38,16 @@ The default detector is `TracerSparsityDetector()` from the package `SparseConne
 Prior to version 0.8.0, the default detector was `SymbolicSparsityDetector()` from `Symbolics.jl`.
 
 - A **`coloring_algorithm`** must be of type `SparseMatrixColorings.GreedyColoringAlgorithm`.
-The default algorithm is `GreedyColoringAlgorithm{:direct}()` from the package `SparseMatrixColorings.jl`.
+The default algorithm is `GreedyColoringAlgorithm{:direct}()` for `SparseADJacobian` and `SparseADHessian`, while it is `GreedyColoringAlgorithm{:substitution}()` for `SparseReverseADHessian`.
+These algorithms are available in the package `SparseMatrixColorings.jl`.
+
+The `GreedyColoringAlgorithm{:direct}()` performs column coloring for Jacobians and star coloring for Hessians.
+In contrast, `GreedyColoringAlgorithm{:substitution}()` applies acyclic coloring for Hessians.
+Substitution coloring results in fewer colors and thus requires fewer directional derivatives to recover all non-zeros of the sparse Hessian.
```
Comment (tmigot) [inline docs/src/sparse.md:46]:
```suggestion
The `:substitution` coloring mode usually finds fewer colors than the `:direct` mode and thus fewer directional derivatives are needed to recover all non-zeros of the sparse Hessian.
```

File: docs/src/sparse.md [hunk 3]
```diff
@@ -38,7 +38,16 @@ The default detector is `TracerSparsityDetector()` from the package `SparseConne
...
+However, it necessitates storing the entire compressed sparse Hessian.
+On the other hand, direct coloring only requires storing one column of the compressed Hessian at a time.
+
+If the Hessian is highly ill-conditioned, direct coloring can be preferable because it avoids the need for triangular solves to recover the non-zeros from the compressed Hessian.
```
Comment (tmigot) [inline docs/src/sparse.md:50]:
```suggestion
The `:direct` coloring mode is numerically more stable and may be preferable for highly ill-conditioned Hessian as it doesn't require solving triangular systems to compute the non-zeros from the compressed Hessian.
```

**Review by tmigot — CHANGES_REQUESTED** [review]

**Comment by amontoison** [comment]
Thanks @tmigot for the review!

**Pull request merged**